### DISCUSSION TITLE
How can I set something to run after all plugins have been loaded?

### DISCUSSION THREAD

**Discussion by volatxs** [wbthomason/packer.nvim#1070] [Q&A] [answered]
I have multiple colorscheme plugins (namely onedark.nvim, gruvbox and catppuccin). I would like to specify which colorscheme to use but this doesn't seem to go too well. The thing is that I configure catppuccin using the "config" key:

```lua
use {
    'catppuccin/nvim',
    as = 'catppuccin',
    config = get_plugconfig_str("catppuccin")
}
```

And then somewhere else I do `colorscheme catppuccin`, but it so happens that the colorscheme is set before loading its config, making it use all the default values and effectively ignoring my configuration unless I manually do something like `colorscheme catppuccin` later.

Is there something (an autocommand, perhaps) that I can use to run something after all the plugins are loaded? In this case I would run the `colorscheme x` command after I was sure the colorschemes are loaded and configured.

(Detail: get_plugconfig_str(x) is just a function that returns a string like `'require("plugconfigs.x")'`.)

**Comment by nat-418** [comment, marked as answer]
First, you should read the [manual on how Neovim startup works](https://neovim.io/doc/user/starting.html). Next, you have to think about how are you loading these plugins. My setup looks like this:

```lua
-- Main Neovim configuration file
-- ~/.config/nvim/init.lua
require('plugins')         -- ~/.config/nvim/lua/pluigins.lua  - Packer config
require('colors')          -- ~/.config/nvim/lua/colors.lua     -  Setting color theme, etc.
require('keymaps')     -- ~/.config/nvim/lua/keymaps.lua - etc.
-- ...etc.
```

Another thing to look at is your `config` declaration. Usually I would write mine like this:

```lua
use({
  'user/plugin',
  config = function()
    require('plugin').setup({
      do = 'something'
    })
  end
})
```

That anonymous callback function will get called once the plugin is loaded and *then* do all the configuration.

**Reply by volatxs** [reply]
Thank you very much for the response. I'll look into the Neovim docs for starting. I think I've tried simply having the "colorscheme" call happen after my plugin setup and I think it does stuff asynchronously or something, but I'll make sure to try again. As for how my config works, you can pass a string to the `config` key and it'll interpret it as lua code to run. I have a folder at `(config dir)/lua/plugconfigs/` where I store plugin-specific configuration files (e.g. I have toggleterm and its configuration lives in the file `(config dir)/lua/plugconfigs/toggleterm.lua`, so by passing a string like `"require("plugconfigs.toggleterm")"` to the `config` key it calls it. It's the same as doing `function() require("plugconfigs.toggleterm") end` but shorter.

**Reply by volatxs** [reply]
Well, it seems it worked for me!
```lua
require("init.general")   -- general options | init/general.lua
require("init.plug")      -- plugin loading  | init/plug.lua
require("init.mappings")  -- mappings        | init/mappings.lua

require("init.folding")  -- folding specials | init/folding.lua

vim.cmd("colorscheme catppuccin")
```
